test/e2e-helm/e2e_suite.go

```diff
@@ -0,0 +1,211 @@
+				"--plugins", "helm.operator-sdk.io/v1",
+				"--project-version", "3-alpha",
+				"--domain", tc.Domain)
+			Expect(err).Should(Succeed())
+
+			By("creating an API definition")
+			err = tc.CreateAPI(
+				"--group", tc.Group,
+				"--version", tc.Version,
+				"--kind", tc.Kind)
+			Expect(err).Should(Succeed())
+
+			By("replace Dockerfile to use the dev tag")
+			version := strings.TrimSuffix(version.Version, "+git")
+			testutils.ReplaceInFile(filepath.Join(tc.Dir, "Dockerfile"), version, "dev")
+
+			By("building the operator image")
+			err = tc.Make("docker-build", "IMG="+tc.ImageName)
+			Expect(err).Should(Succeed())
+
+			kubectx, err := tc.Kubectl.Command("config", "current-context")
+			Expect(err).Should(Succeed())
+
+			if strings.Contains(kubectx, "kind") {
+				By("loading the operator image into the test cluster")
+				err = tc.LoadImageToKindCluster()
+				Expect(err).Should(Succeed())
+			}
+
+			By("install CRDs")
+			err = tc.Make("install")
+			Expect(err).Should(Succeed())
+
+			By("deploying the controller manager")
+			err = tc.Make("deploy", "IMG="+tc.ImageName)
+			Expect(err).Should(Succeed())
+
+			By("ensuring the controller-manager pod is running as expected")
+			verifyControllerUp := func() error {
+				// Get pod name
+				podOutput, err := tc.Kubectl.Get(
+					true,
+					"pods", "-l", "control-plane=controller-manager",
+					"-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}"+
+						"{{ \"\\n\" }}{{ end }}{{ end }}")
+				Expect(err).NotTo(HaveOccurred())
+				podNames := kbtestutils.GetNonEmptyLines(podOutput)
+				if len(podNames) != 1 {
+					return fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+				}
+				controllerPodName = podNames[0]
+				Expect(controllerPodName).Should(ContainSubstring("controller-manager"))
+
+				// Validate pod status
+				status, err := tc.Kubectl.Get(
+					true,
+					"pods", controllerPodName, "-o", "jsonpath={.status.phase}")
+				Expect(err).NotTo(HaveOccurred())
+				if status != "Running" {
+					return fmt.Errorf("controller pod in %s status", status)
+				}
+				return nil
+			}
+			Eventually(verifyControllerUp, time.Minute, time.Second).Should(Succeed())
+
+			By("granting permissions to access the metrics and read the token")
+			_, err = tc.Kubectl.Command(
+				"create", "clusterrolebinding", fmt.Sprintf("metrics-%s", tc.TestSuffix),
+				fmt.Sprintf("--clusterrole=e2e-%s-metrics-reader", tc.TestSuffix),
+				fmt.Sprintf("--serviceaccount=%s:default", tc.Kubectl.Namespace))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("getting the token")
+			b64Token, err := tc.Kubectl.Get(true, "secrets", "-o=jsonpath={.items[0].data.token}")
+			Expect(err).NotTo(HaveOccurred())
+			token, err := base64.StdEncoding.DecodeString(strings.TrimSpace(b64Token))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(token)).To(BeNumerically(">", 0))
+
+			By("creating a pod with curl image")
```

Comment:
We might want to do this after deploying the CR so that we can test for the existence of certain metrics we expect to see (e.g. the updating resource_created_at_seconds metrics).